src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -142,21 +131,25 @@ public abstract class PeerConnector : IPeerConnector
         }
 
         /// <summary>Constructor used for unit testing.</summary>
-        protected PeerConnector(NodeSettings nodeSettings, IPeerAddressManager peerAddressManager)
+        protected PeerConnector(ILoggerFactory loggerFactory, NodeSettings nodeSettings, IPeerAddressManager peerAddressManager)
         {
             Guard.NotNull(nodeSettings, nameof(nodeSettings));
             Guard.NotNull(peerAddressManager, nameof(peerAddressManager));
 
+            this.connectedPeers = new NetworkPeerCollection();
+            this.loggerFactory = loggerFactory;
```

Comment:
`NodeLifetime` must be injected

Reply:
Same here, these "unit test" constructors has been dropped in #891